### PR TITLE
some fixes in subcommand handling: allow defaulting to new commands and reorder them for clarity

### PR DESCRIPTION
**Pull request by inaky-intc**

### CODE REVIEW

File: tcf [hunk 1]
```diff
@@ -555,10 +527,74 @@ if __name__ == "__main__":
...
     ap.set_defaults(func = healthcheck)
 
+    tcfl.tc.target_ext_certs.cmdline_setup(arg_subparsers)
+
+
+    # advanced commands
+
+    ap = arg_subparsers.add_parser("cookies",
+                                   help = "Show logging cookies (to feed"
+                                   " into curl, etc) maybe only for one server")
```
Comment (bsantana-intc) [inline tcf:537]: pretty cool

File: tcf [hunk 2]
```diff
@@ -555,10 +527,74 @@ if __name__ == "__main__":
...
+    ap.set_defaults(func = _target_disable)
+
+
+    ap = arg_subparsers.add_parser(
+        "patch", help = "Store target information in the server")
+    ap.add_argument("target", metavar = "TARGET", action = "store",
+                    default = None, help = "Target's name")
+    ap.add_argument("data", metavar = "KEY=JSON-VALUE", nargs = "*",
+                    default = None, help = "Data items to store; if"
+                    " none, specify a JSON dictionary over stdin")
```
Comment (bsantana-intc) [inline tcf:571]: what is the diff with `tcf property-set`?

File: tcf [hunk 3]
```diff
@@ -285,6 +286,14 @@ def _cookies(args):
 
 if __name__ == "__main__":
     tcfl.tc.version = commonl.version_get(tcfl, "tcf")
+
+    if "TCF_NEW_COMMANDS" in os.environ:
+        tcfl.ui_cli.commands_old_suffix = "-old"
+        tcfl.ui_cli.commands_new_suffix = ""
+    else:
+        tcfl.ui_cli.commands_old_suffix = ""
+        tcfl.ui_cli.commands_new_suffix = "2"
```
Comment (bsantana-intc) [inline tcf:295]: basically, if we have that variable in the env, we wont have to specify `2` in every command and if we want to use the old ones we add `-old`. This is cool, where should we have this documented?

**Review by bsantana-intc — APPROVED** [review]

File: tcf (hunk 2 above)
Comment (inaky-intc) [reply tcf:571]: patch allows you to specify a whole json with multiple fields; property-set is just one property and value

File: tcf (hunk 3 above)
Comment (inaky-intc) [reply tcf:295]: yap; I am hesitant on documenting it for the time being, more of a developer feature?

File: tcfl/ui_cli_power.py [hunk 4]
```diff
@@ -170,12 +170,13 @@ def _state_to_str(state):
...
-            commonl.data_dump_recursive(r, prefix = target.fullid)
+            commonl.data_dump_recursive(r, prefix = target.id)
 
         else:  # verbosity >= 2:
-            r = dict(state = state, substate = substate,
-                     components = components)
-            print(json.dumps(r, skipkeys = True, indent = 4))
+            r = { target.id: dict(state = state, substate = substate,
+                                  components = components) }
+            json.dump(r, sys.stdout, skipkeys = True, indent = 4)
```
[the remark this answers is not in the document]
Comment (inaky-intc) [reply tcfl/ui_cli_power.py:178]: json.dump() will dump the json tree strating to stdout (sys.stdout), the second arg -- the prev version was kinda dumb, when I still didn't know json.dump() was there :)

[the remark this answers is not in the document]
Comment (alkaajit) [reply tcfl/ui_cli_power.py:178]: Ah ok! Makes sense!

**Pull request merged**